src/social-media-post/models/social-media-post.entity.ts

```diff
@@ -0,0 +1,20 @@
+import { Column, Entity, ManyToOne, PrimaryGeneratedColumn } from 'typeorm';
+import { UserEntity } from '../../user/models/user.entity';
+
+@Entity('social_posts')
+export class SocialMediaPostEntity {
```

Comment:
I'd also add a `deleted_at` field:

- If this field is not `null`, it's public and viewable via the app.
- If it contains a timestamp, we consider it deleted (for any reason: abuse, GDPR, user request, etc.)

Reply:
Good point, will add.